sources/window/error_source.go

```diff
@@ -165,7 +167,7 @@ func (e *errorHealthCheckSource) getFailureResult() health.HealthCheckResult {
 	}
 	healthCheckResult := health.HealthCheckResult{
 		Type:    e.checkType,
-		State:   health.New_HealthState(health.HealthState_ERROR),
+		State:   health.New_HealthState(e.healthState),
```

Comment:
Doesn't this cause an issue if `e.healthState` hasn't been set and is empty/blank? Should the default state be `health.HealthState_ERROR` and then only changed if `e.healthState` is non-empty?

Reply:
This would only be possible if you had access to the private `errorHealthCheckSource`. In each constructor the first step is to build the default config, which includes setting `e.healthState = health.HealthState_ERROR`, and then we return the private `errorHealthCheckSource`. So this should be fully covered.